### PR TITLE
feat: protected routing bb-12

### LINKED ISSUE #12: feat: Protected routing
- [x]  Create ProtectedRoute wrapper component
- [x]  Redirect not signed in users from protected screens to Sign In page

### PR DESCRIPTION
**Pull request by marharita08**
## Created
- protected route component

## Updated
- components.tsx to export protected route component
- index.tsx to wrap root path in protected route
- auth.slice.ts added user to state type and to initial state

### CODE REVIEW

File: apps/frontend/src/libs/components/protected-route/protected-route.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { useAppSelector } from "~/libs/hooks/hooks.js";
+
+const ProtectedRoute: React.FC = () => {
+	const authStatus = useAppSelector(({ auth }) => auth.dataStatus);
+
+	if (authStatus === DataStatus.FULFILLED) {
+		return <Outlet />;
+	}
+
+	return <Navigate replace to={AppRoute.SIGN_IN} />;
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/protected-route.tsx:13]: You shouldn't check the status. You should check whether the user is present or not

File: apps/frontend/src/libs/components/protected-route/protected-route.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Navigate, Outlet } from "react-router-dom";
+
+import { AppRoute, DataStatus } from "~/libs/enums/enums.js";
+import { useAppSelector } from "~/libs/hooks/hooks.js";
+
+const ProtectedRoute: React.FC = () => {
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/protected-route.tsx:6]: This component should receive `component` and `redirectTo` as props. You will either display that component or redirect to `redirectTo` screen

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by WilhelmWesser — APPROVED** [review]

**Review by WilhelmWesser — APPROVED** [review]

**Review by revenets — APPROVED** [review]

File: apps/frontend/src/libs/components/protected-route/protected-route.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Navigate } from "react-router-dom";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/protected-route.tsx:1]: Should be imported from ~/libs/components/components.js

File: apps/frontend/src/libs/components/protected-route/protected-route.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import { ValueOf } from "shared";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/protected-route.tsx:2]: Should be imported from ~/libs/types/types.js

File: apps/frontend/src/modules/auth/slices/auth.slice.ts [hunk 5]
```diff
@@ -2,15 +2,18 @@ import { createSlice } from "@reduxjs/toolkit";
 
 import { DataStatus } from "~/libs/enums/enums.js";
 import { type ValueOf } from "~/libs/types/types.js";
+import { UserSignUpResponseDto } from "~/modules/users/libs/types/types.js";
```
Comment (fshabanov) [inline apps/frontend/src/modules/auth/slices/auth.slice.ts:5]:
```suggestion
import { UserSignUpResponseDto } from "~/modules/users/users.js";
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Comment by fshabanov** [comment]
Merge the `main` branch into yours

**Comment by fshabanov** [comment]
Merge it again, please :)

**Review by fshabanov — APPROVED** [review]

**Pull request merged**